Settings for page or browser context.

terminalkitten opened the pull request:
Open for discussion, but the general idea is to add `settings` dict to `pdf.configuration`. 
This way we can pass settings to page, browser and more.

For example to handle issue [How to set page.emulateMedia('screen')? #3](https://github.com/shivanshs9/pdfgen-python/issues/3)

      import pdfgen
      
      pdfgen.configuration(settings={'emulateMedia': 'screen'})


Another example:

      import pdfgen
      
      pdfgen.configuration(settings={'setBypassCSP': True})


I think there should be a list of page & browser options that are relevant for `Page.pdf`:

- Page.emulateMedia
- Page.setBypassCSP
- Page.setCookie

To wait for the page
- Page.goto{waitUntil}

To block / intercept certain requests
- Page.setRequestInterception
- Page.on('request')

Commit:
added settings to pdf.config, added emulateMedia setting for page

shivanshs9 reviewed (changes requested):
Since `settings` is quite similar to `options`, it may lead to confusion when using it. I suggest using the name of `pyppeteer_tweaks` or something similar since that would clarify it's for tweaking pyppeteer.

terminalkitten commented:
> Since `settings` is quite similar to `options`, it may lead to confusion when using it. I suggest using the name of `pyppeteer_tweaks` or something similar since that would clarify it's for tweaking pyppeteer.

Good point, this would definitly confuse users. What about just `pyppeteer `, tweaks also suggest otherwise imo:
   
    import pdfgen
  
    pdfgen.configuration(pyppeteer={'emulateMedia': 'screen'})


Also added extra options for Pyppeteer Page.goto{options} and Browser.args:

    pdfgen.configuration(pyppeteer={
       'browserArgs': ['--webkit-print-color-adjust'],
       'pageOptions': {'waitUntil': 'networkidle2'},
    })


Added new settings for:

- setBypassCSP
- setRequestInterception
- setCookie

Commit:
added basic test for settings

Commit:
pass on pyppeteer settings, always cofigure empty options

Commit:
apply multiple settings from pyppeteer configuration, added pageOptions and addtional browserArgs
